lib/rspec/core/ordering.rb

```diff
@@ -66,6 +66,20 @@ def order(list)
         end
       end
 
+      class Priority
+        def initialize(files=[])
+          @files = files
+        end
+
+        def order(list)
+          priority_files = list.select do |item|
```

Comment:
[`partition`](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-partition)?

Reply:
I considered using partition but found it to be too slow in some cases. I will do more research on this.

Reply:
Done!